service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -210,49 +222,103 @@ public void run() {
             phaser.arrive();
             while (true) {
                 try {
-                    SampleMetadata sampleMetadata = sampleUpdateQueue.poll(100, TimeUnit.MILLISECONDS);
+                    SampleMetadata sampleMetadata = researchSampleUpdateQueue.poll(
+                            100, TimeUnit.MILLISECONDS);
                     if (sampleMetadata != null) {
                         MetadbSample existingSample = sampleService.getResearchSampleByRequestAndIgoId(
                                 sampleMetadata.getIgoRequestId(), sampleMetadata.getPrimaryId());
                         if (existingSample == null) {
-                            LOG.info("Sample metadata does not already exist - persisting to db: "
+                            LOG.info("research Sample metadata does not already exist - persisting to db: "
                                     + sampleMetadata.getPrimaryId());
                             // handle and persist new sample received
                             MetadbSample sample = SampleDataFactory
                                     .buildNewResearchSampleFromMetadata(sampleMetadata.getIgoRequestId(),
                                             sampleMetadata);
-                            sampleService.saveSampleMetadata(sample);
-                            LOG.info("Publishing metadata history for new sample: "
+                            sampleService.saveMetadbSample(sample);
```

Comment:
can you remind me why we had to change the method to save a `MetadbSample` instead of `SampleMetadata`? Was there something wrong saving the `SampleMetadata` logic?